### ISSUE TITLE
gaps between cells

### ISSUE THREAD

**Issue by Mobyrr** [kozakdenys/qr-code-styling#235]
in the version 1.7.2 and below (not on 1.5.0), there is gaps of 1px between the cells
![image](https://github.com/user-attachments/assets/f389cee7-352e-46ca-827c-5c8ab83dfe8f)
on some size multiples, theses gaps are not visible
theses gaps are also visible with other styles
![image](https://github.com/user-attachments/assets/dd2961be-7ceb-46cc-a585-1adb8ffea3ac)

**Comment by Mobyrr** [comment]
I imagine that the problem may be complicated to reproduce
A part of my code (surely not helping) :
```ts
onMounted(() => {
  watch([props, size, margin], () => {
    if (!props.value || !show.value) return
    nextTick(() => {
      canvasElement.value!.innerHTML = ''
      qrCode = new QRCodeStyling({
        data: props.value,
        width: parseInt(size.value),
        height: parseInt(size.value),
        margin: parseInt(margin.value),
        dotsOptions: {
          type: 'extra-rounded',
          color: '#009EE2'
        },
        cornersSquareOptions: {
          type: 'extra-rounded',
          color: '#009EE2'
        },
        imageOptions: {}
      })
      qrCode.append(canvasElement.value)
    })
  }, { immediate: true })
})
```

The problem occurs on chromium browsers as well as on firefox on a windows without hardware acceleration.

**Comment by kozakdenys** [comment]
Hi @Mobyrr, in the process of fixing it. I will create release soon

**Comment by nullifye** [comment]
i come across this issue for my project when using mobile browser (chrome android). it's working fine on desktop. is this issue was fixed on 1.8.4?

[test link](afb.my/jQLzoZ.qr)

![6242360697447104455](https://github.com/user-attachments/assets/7136c99f-5722-4b21-adc1-c6a2b1ce71f8)

**Comment by nullifye** [comment]
i try to use **roundSize** option and it fix the issue on mobile browser, but another problem arise on desktop browser; invisible box covered some part or qr code.

so i decided to manually add **shape-rendering** to the svg if and only if user is on mobile after detecting device type.

my problem is fixed for now.
